### PR TITLE
[codex] Drop sentence-transformers dependency

### PR DESCRIPTION
**Pull request by ranieri**
## Summary

- replace the SentenceTransformers finetuning wrapper with a native PyTorch embedding module around `ASMBertModel` and `ASMTokenizer`
- rewrite finetuning to use native label batching, semi-hard triplet loss, triplet evaluation, checkpointing, and standard `save_pretrained()` outputs
- remove `sentence-transformers` from project dependencies and refresh asmtransformers/citatio lockfiles
- update tests and docs for the native finetuning API and intentional `build_finetuning_model()` return-type break

## Impact

Finetuned checkpoints are now saved as regular `ASMBertModel` checkpoints plus tokenizer files and are loadable through `ASMEmbedder`. Old SentenceTransformer checkpoint metadata and APIs such as `.fit()`, `.save()`, and module indexing are no longer supported.

## Validation

- `cd asmtransformers && pdm run check`
- `cd asmtransformers && pdm run test` (76 passed)
- `cd citatio && pdm run check`
- `cd citatio && pdm run test` (14 passed)

### CODE REVIEW

**Comment by ranieri** [comment]
Mainly intended to be food for discussion
